### PR TITLE
Make worker threads and timeouts configurable.

### PR DESCRIPTION
**Pull request by blinsay**
Allow configuring the number of worker threads and handoff timeout with `NSNCD_WORKER_COUNT` and `NSNCD_HANDOFF_TIMEOUT` to tune cpu usage on busy hosts.

This also changes the default number of workers from 256 to 8. We may want to roll that out separately.

### CODE REVIEW

**Review by leifwalsh — APPROVED** [review]

File: src/main.rs [hunk 1]
```diff
@@ -75,15 +74,24 @@ fn main() -> Result<()> {
 
     let logger = slog::Logger::root(drain, slog::o!());
 
+    let worker_count = env_usize("NSNCD_WORKER_COUNT", 8);
+    let handoff_timeout = Duration::from_secs(env_usize("NSNCD_HANDOFF_TIMEOUT", 3) as u64);
```
Comment (domenkozar) [inline src/main.rs:78]: 3 seconds is way too low, at cachix.org we've been experiencing problems due to this. Can we make it 10?

Comment (blinsay) [reply src/main.rs:78]: Are you able to tune the timeout with NSNCD_HANDOFF_TIMEOUT?

Comment (domenkozar) [reply src/main.rs:78]: Yes, however I'd much prefer a fix upstream.

https://github.com/NixOS/nixpkgs/pull/290355

Comment (blinsay) [reply src/main.rs:78]: Can you open an issue where we can discuss this? Tell us more about what kind of load you're putting on nsncd at Cachix, how often you see the timeouts and on which calls, and what your upstream NSS calls are (e.g. are you reaching out to LDAP or some other networked service).

**Pull request merged**